src/main/java/xyz/nucleoid/plasmid/util/BucketFind.java

```diff
@@ -0,0 +1,43 @@
+package xyz.nucleoid.plasmid.util;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.math.BlockPos;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class BucketFind {
+    /**
+     * Finds any near blocks and puts them in a set.
+     *
+     * @param pos      The position of the first block
+     * @param blockTag The tag of the blocks that can be accepted in the set
+     */
+    public static Set<BlockPos> find(ServerWorld world, BlockPos pos, Tag<Block> blockTag) {
+        Set<BlockPos> positions = new HashSet<>();
+        positions.add(pos);
+        searchAround(world, pos, positions, blockTag);
+        return positions;
+    }
+
+    private static void searchAround(ServerWorld world, BlockPos pos, Set<BlockPos> positions, Tag<Block> blockTag) {
+        for(int x = -1; x <= 1; x++) {
+            for(int z = -1; z <= 1; z++) {
+                for(int y = -1; y <= 1; y++) {
```

Comment:
I think this should only search the 6 blocks directly connected to `pos`. Iterating over `Direction.values()` instead should work better.

Reply:
Fixed in 3145459e790e7740817617dd1e6999ad831ae704